3.preprocess-features/preprocess-features.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### Import Libraries
+
+# In[1]:
+
+
+import pathlib
+from pycytominer.cyto_utils import DeepProfiler_processing
+import pandas as pd
+import numpy as np
+import joblib
+
+import importlib
+preprocessUtils = importlib.import_module("preprocess-features-utils")
+
+
+# ### Specify File/Folder paths
+
+# In[2]:
+
+
+intermediate_files_dir = pathlib.Path("intermediate_files/")
+intermediate_files_dir.mkdir(parents=True, exist_ok=True)
+
+annotations_path = pathlib.Path("../0.image-download/manifest/idr0080-screenA-annotation.csv")
+
+DP_project_path = pathlib.Path("/media/roshankern/63af2010-c376-459e-a56e-576b170133b6/data/cell-health-nuc-DP")
+features_output_dir = pathlib.Path(f"{DP_project_path}/outputs/efn_pretrained/features")
+original_index_csv_path = pathlib.Path(f"{DP_project_path}/inputs/metadata/index.csv")
+
+output_path = pathlib.Path("/media/roshankern/63af2010-c376-459e-a56e-576b170133b6/data/cell-health-nuc-per-plate-normalized/")
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# ### Convert index.csv to int values
+
+# In[3]:
+
+
+index_csv_path = pathlib.Path(f"{intermediate_files_dir}/index.csv")
+# convert string values to int where possible (ex with well number)
+# necessary because pycytominer reads index.csv file with `dtype=str`
+original_index_csv = pd.read_csv(original_index_csv_path)
+original_index_csv.to_csv(index_csv_path)
+
+
+# ### Find negative controls for each plate, derive scalers from these negative controls, normalize entire plate with scaler, save normalized plate
+
+# In[4]:
+
+
+# get list of unique plates
+plates = pd.read_csv(index_csv_path)["Metadata_Plate"].unique().tolist()
+
+for plate in plates:
```

Comment:
no content changes, but I do suggest adding some white space (empty lines).

This is really dense code, white space will enable someone to see what's going on with higher clarity.

Reply:
Done in https://github.com/WayScience/cell-health-data/pull/7/commits/f2881f98973b71dc4d4e255c378008e6846d1e1b